RestClient.cpp

```diff
@@ -281,26 +263,26 @@ int RestClient::readResponse(String* response) {
                     }
 
                     if (c == '\n') {
-                        // you're starting a new line
+                        // your starting a new line
                         currentLineIsBlank = true;
                     }
                     else if (c != '\r') {
                         // you've gotten a character on the current line
                         currentLineIsBlank = false;
                     }
+
+                	rawLastResponse = rawLastResponse + c;
                 }
             }
         }
         HTTP_DEBUG_PRINT("HTTPS client closed \n");
     }else {
-        while (client.connected()) {
-            HTTP_DEBUG_PRINT(".");
+        while (client.connected() || client.available()) {
```

Comment:
I don't think you want this extra `|| client.available()` here.  While it should never be true if connected() is false it incorrectly tells other developers that we should execute this block even if the client is not connected.